src/components/shapes/draw.js

```diff
@@ -129,14 +138,21 @@ function drawOne(gd, index) {
             opacity = gd._fullLayout.activeshape.opacity;
         }
 
-        var path = shapeLayer.append('path')
+        var shapeGroup = shapeLayer.append('g')
+            .classed('shape-group', true)
+            .attr({ 'data-index': index });
+
+        var path = shapeGroup.append('path')
             .attr(attrs)
             .style('opacity', opacity)
             .call(Color.stroke, lineColor)
             .call(Color.fill, fillColor)
             .call(Drawing.dashLine, lineDash, lineWidth);
 
-        setClipPath(path, gd, options);
+        setClipPath(shapeGroup, gd, options);
+
+        // Draw label, if present
```

Comment:
```suggestion
        // Draw or clear the label
```
Or you could simply remove the comment :)